csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.shared.cs

```diff
@@ -305,7 +305,7 @@ static NativeMethods()
             DOrtGetApi OrtGetApi = (DOrtGetApi)Marshal.GetDelegateForFunctionPointer(OrtGetApiBase().GetApi, typeof(DOrtGetApi));
 
             // TODO: Make this save the pointer, and not copy the whole structure across
-            api_ = (OrtApi)OrtGetApi(4 /*ORT_API_VERSION*/);
+            api_ = (OrtApi)OrtGetApi(14 /*ORT_API_VERSION*/);
```

Comment:
is this supposed to match the current value in the C header?

Reply:
I think this should match the functions we expect to be provided by OrtApi. It hasn't been updated for years which I think is incorrect. If I called an old version of ORT asking for API Version 4 it would break if you tried to use anything from a newer version. 

Really this should be 15 but I'm not sure if that would break any test setups we have so I went for the safer 14 for now. better than 4 at least. can try and update to match the latest ORT version post release (i.e. change to 16 when ORT_API_VERSION on main changes).